src/blob_gc_job.cc

```diff
@@ -341,7 +360,11 @@ bool BlobGCJob::DiscardEntry(const Slice& key, const BlobIndex& blob_index) {
     fprintf(stderr, "GetImpl err, status:%s\n", s.ToString().c_str());
     abort();
   }
+  RecordTick(stats_, BLOB_DB_BYTES_READ, key.size() + index_entry.size());
   if (s.IsNotFound() || !is_blob_index) {
+    RecordTick(stats_, BLOB_DB_GC_NUM_KEYS_OVERWRITTEN);
```

Comment:
Both `DoSample()` and `DoRunGC()` call `DiscardEntry()`, and we shouldn't log these counter while sampling. Move the logic to `DoRunGC()`? Same for `BLOB_DB_GC_XXX_RELOCATED` counters below.

Reply:
if do it in `DoRunGC()`, we can't distinguish whether it is overwritten or relocated. Maybe add a bool to decide whether need to count metrics.